config/http_config.go

```diff
@@ -148,6 +152,7 @@ type Authorization struct {
 	Type            string `yaml:"type,omitempty" json:"type,omitempty"`
 	Credentials     Secret `yaml:"credentials,omitempty" json:"credentials,omitempty"`
```

Comment:
not(non blocking): Wild idea, but what stops us to use Credentials file as a ref. Or *File? 

I guess potential confusion and lack of ability for users to use inline somewhere, file somewhere and X secret provider in another place on one deployment? 🤔

Reply:
Prometheus uses these objects to load the YAML configurations. Prometheus has no way of knowing whether you actually want a file or a ref. If it somehow knew, you could add the files to the secret manager under a special "file secret provider". If we didn't use these objects for unmarshaling, your idea would work.